### ISSUE TITLE
Pydantic Issue

### ISSUE THREAD

**Issue by rishacha** [langchain-ai/langchain-ibm#18]
## Details

The following error occurs when trying to import `from langchain_ibm import WatsonxLLM`
Python version - **Python 3.11.9**

```py
class WatsonxLLM(BaseLLM):
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 224, in __new__
    complete_model_class(
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 573, in complete_model_class
    schema = cls.__get_pydantic_core_schema__(cls, handler)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/main.py", line 668, in __get_pydantic_core_schema__
    return handler(source)
           ^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_schema_generation_shared.py", line 83, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 655, in generate_schema
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 924, in _generate_schema_inner
    return self._model_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 739, in _model_schema
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 739, in <dictcomp>
    {k: self._generate_md_field_schema(k, v, decorators) for k, v in fields.items()},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 1115, in _generate_md_field_schema
    common_field = self._common_field_schema(name, field_info, decorators)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 1308, in _common_field_schema
    schema = self._apply_annotations(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 2107, in _apply_annotations
    schema = get_inner_schema(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_schema_generation_shared.py", line 83, in __call__
    schema = self._handler(source_type)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 2088, in inner_handler
    schema = self._generate_schema_inner(obj)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 929, in _generate_schema_inner
    return self.match_type(obj)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 1029, in match_type
    return self._match_generic_type(obj, origin)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 1058, in _match_generic_type
    return self._union_schema(obj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 1378, in _union_schema
    choices.append(self.generate_schema(arg))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 657, in generate_schema
    metadata_js_function = _extract_get_pydantic_json_schema(obj, schema)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/test/code/pilots/iss/slack-tririga/.venv/lib/python3.11/site-packages/pydantic/_internal/_generate_schema.py", line 2447, in _extract_get_pydantic_json_schema
    raise PydanticUserError(
pydantic.errors.PydanticUserError: The `__modify_schema__` method is not supported in Pydantic v2. Use `__get_pydantic_json_schema__` instead in class `SecretStr`.

For further information visit https://errors.pydantic.dev/2.9/u/custom-json-schema
```

**Comment by MateuszOssGit** [comment]
Hi @rishacha 
Could you please provide me your version of `langchain-core`?

```
pip show langchain-core

```

**Comment by MateuszOssGit** [comment]
It is related with `0.3.0` version of `langchain-core`
For now `langchain_ibm` have in dependencies
```
langchain-core = ">=0.2.2,<0.3"
```

**Comment by rishacha** [comment]
#22 - Fixes this I guess?

**Comment by MateuszOssGit** [comment]
@rishacha It should be resolved with 0.2.0 (and above) version of `langchain-ibm` package.

**Comment by rishacha** [comment]
It's resolved now - this is how my `pyproject.toml` dependencies look like atm - 

```toml
dependencies = [
    "langchain-core>=0.3",
    "langchain-ibm>=0.1.12",
    "torch>=2.3.1",
    "langchain-huggingface>=0.0.3",
    "openpyxl>=3.1.5",
    "faiss-cpu>=1.8.0.post1",
    "ibm-watsonx-ai>=1.1.9",
    "fastapi[standard]>=0.114.2",
    "pydantic-settings>=2.5.2",
    "pydantic>=2.9.1",
    "slack-sdk>=3.32.0",
    "requests>=2.32.2",
    "langchain-community>=0.2.17",
]
```
